mod/beacon/validator/propose.go

```diff
@@ -102,46 +83,15 @@ func (s *Service[
 		return blk, sidecars, ErrNilPayload
 	}
 
-	// If we get returned a nil blobs bundle, we should return an error.
-	blobsBundle := envelope.GetBlobsBundle()
-	if blobsBundle == nil {
-		return blk, sidecars, ErrNilBlobsBundle
-	}
-
-	// Set the KZG commitments on the block body.
-	body.SetBlobKzgCommitments(blobsBundle.GetCommitments())
-
-	depositIndex, err := st.GetEth1DepositIndex()
-	if err != nil {
-		return blk, sidecars, ErrNilDepositIndexStart
-	}
-
-	// Dequeue deposits from the state.
-	deposits, err := s.bsb.DepositStore(ctx).GetDepositsByIndex(
-		depositIndex,
-		s.chainSpec.MaxDepositsPerBlock(),
+	// Create a new empty block from the current state.
+	blk, err = s.getEmptyBeaconBlockForSlot(
+		st, requestedSlot,
```

Comment:
I don't think instantiating this later is necessarily the same because of how retrieveExecutionPayload is handled: https://github.com/berachain/beacon-kit/blob/f010656737a32bfc59f5dbc460979f7c7ef6393a/mod/beacon/validator/propose.go#L256

Reply:
@ocnc tks. I've just fixed